src/extension/op_def.rs

```diff
@@ -143,26 +141,36 @@ pub struct OpDef {
     lower_funcs: Vec<LowerFunc>,
 }
 
-impl TypeParametrised for OpDef {
-    type Concrete = OpaqueOp;
-
-    fn params(&self) -> &[TypeParam] {
-        self.params()
-    }
-
-    fn name(&self) -> &SmolStr {
-        self.name()
-    }
-
-    fn extension(&self) -> &ExtensionId {
-        self.extension()
-    }
-}
-
 impl OpDef {
-    /// Check provided type arguments are valid against parameters.
-    pub fn check_args(&self, args: &[TypeArg]) -> Result<(), SignatureError> {
-        self.check_args_impl(args)
+    /// Check provided type arguments are valid against [ExtensionRegistry],
+    /// against parameters, and that no type variables are used as static arguments
```

Comment:
Hmmm, I could link to `[Self::params()]`, but I also wonder if we should keep that, or switch to `static_params()` and `substitutable_params(&[TypeArg])` (where the latter has to call the custom binary function)